### PR TITLE
CI: Add documentation generation job for FBX CI and a corresponding doc change trigger.

### PR DESCRIPTION
**Pull request by windxu88**
## Purpose of this PR:
Add doc generation job to FBX Exporter package and add a PR trigger for doc only changes.

**JIRA ticket:**
[FBX-496](https://jira.unity3d.com/browse/FBX-496) CI: Add documentation generation job for FBX CI and a corresponding doc change trigger.

### CODE REVIEW

**Comment by windxu88** [comment]
You can find the generated documentation in [the artifacts of this job](https://unity-ci.cds.internal.unity3d.com/job/30517360/artifacts).

File: .yamato/upm-ci.yml [hunk 1]
```diff
@@ -46,6 +46,24 @@ api_doc_validation:
...
 
+# Job to generate documentation for the package
+generate_documentation:
+  name : Generate documentation
+  agent:
+    type: {{ mac_platform.type }}
+    image: {{ mac_platform.image }}
+    flavor: {{ mac_platform.flavor }}
+  commands:
+# Run build script to copy CHANGELOG.md and LICENSE.md into com.unity.formats.fbx package.
```
Comment (vkovec) [inline .yamato/upm-ci.yml:57]: nit: indentation is off here

**Review by vkovec — APPROVED** [review]
looks good I was able to download and view the generated doc

**Review by sebastienduverne — APPROVED** [review]
Nice, thanks, works well!

**Pull request merged**